### ISSUE TITLE
Using face detection possible?

### ISSUE THREAD

**Issue by jaapiyo** [smartcrop/smartcrop.py#11]
What would be an easy way to incorporate face detection using openCV or something similair? The original smartcrop.js library has some examples to do this but I can't find anything similair for the Python-port.

**Comment by davidfischer-ch** [comment]
Hello,

I personally have no time to implement this feature. However, I am ready to help you integrating this criteria to the heuristic.

Thank you.

**Comment by davidfischer-ch** [comment]
Regarding your question, I will investigate it promptly.

**Comment by davidfischer-ch** [comment]
OK https://github.com/jwagner/smartcrop.js/commit/afc15fd5831a890ce7c97fd7a21f4a5df71cfadb is using the alpha channel to store what they call the "boost" value.

- The alpha channel is probably zeroed (default).
- The regions (pixels in alpha channel) with faces are set to a positive value (the weight).
- Finally, the alpha channel / boost is integrated to the [scoring heuristic](https://github.com/jwagner/smartcrop.js/commit/afc15fd5831a890ce7c97fd7a21f4a5df71cfadb#diff-e3e9132e1fc354282b23bac497b41e3bR314).

So in summary, integrating a similar feature requires some boilerplate code.
I am ready to implement such a skeleton hosting an extra criteria to scoring (boost).

Then I will let you, @jaapiyo, implement a particular boost method (e.g. based on OpenCV face detection).

Do you agree with my proposal?

**Comment by Abhi5hekk** [comment]
can anyone help me to integrate face detection?

**Comment by afeezaziz** [comment]
It has been quite some time, so rebump this issue.

@davidfischer-ch is it possible to detect faces now?

**Comment by eeyrw** [comment]
I add face detection to the lib. You may have a try: https://github.com/eeyrw/smartcrop.py
The OpenCV>=4.8.0 is required due to face detection.
